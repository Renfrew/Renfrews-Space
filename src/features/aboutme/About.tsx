import React, { ReactElement } from 'react';
import { Theme, makeStyles, createStyles } from '@material-ui/core/styles';

import { Box, Container, Grid, Paper, Typography } from '@material-ui/core';

import space from '../asserts/Mountains-Outer-Space-Wallpapers.jpg';
import html5 from '../asserts/html5.png';
import css from '../asserts/css-512.png';
import javascript from '../asserts/512px-JavaScript-logo.png';
import typescript from '../asserts/typescript.png';
import reactLogo from '../asserts/react.png';
import reduxLogo from '../asserts/Redux.png';
import aws from '../asserts/aws_icon_146074.png';

const useStyles = makeStyles(({ breakpoints, spacing }: Theme) =>
  createStyles({
    container: {
      minHeight: '100vh',
      backgroundImage: `url(${space})`,
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
      backgroundBlendMode: 'difference',
      backgroundAttachment: 'fixed',
      '& h1,h5,p': {
        color: 'white',
        mixBlendMode: 'difference',
        filter: 'drop-shadow(.05em .05em #cf0c0c)',
      },
    },
    flip: {
      transform: `scale(1.1, -2)`,
    },
    parent: {
      paddingTop: spacing(8),
      marginBottom: spacing(8),
      [breakpoints.down('sm')]: {
        paddingLeft: '10%',
      },
      [breakpoints.up('sm')]: {
        paddingLeft: breakpoints.values.sm * 0.1 + 'px',
      },
    },
    techStack: {
      marginTop: '15px',
      marginBottom: spacing(10),
      width: '90%',
      maxWidth: breakpoints.values.sm,
      marginLeft: 'auto',
      marginRight: 'auto',
      display: 'flex',
      justifyContent: 'space-between',
    },
    techStackLogo: {
      height: '60px',
      width: '60px',
      '& img': {
        height: '60px',
        width: '60px',
      },
    },
    reactLogo: {
      height: '60px',
      width: '60px',
      textAlign: 'center',
      paddingTop: '5px',
      '& img': {
        height: '50px',
        width: '50px',
      },
    },
    reduxLogo: {
      height: '60px',
      width: '60px',
      '& img': {
        height: '60px',
        width: '60px',
      },
    },
    mainContent: {
      textAlign: 'center',
      maxWidth: '50%',
      [breakpoints.down('sm')]: {
        maxWidth: '75%',
      },
      [breakpoints.up('sm')]: {
        fontSize: '1.5em',
      },
      [breakpoints.up('lg')]: {
        maxWidth: '33%',
      },
    },
    ending: {
      height: spacing(10),
    },
  })
);

const techStackIcon: { src: string; alt: string }[] = [
  { src: html5, alt: 'HTML5 icon' },
  { src: css, alt: 'CSS icon' },
  { src: javascript, alt: 'Javascript icon' },
  { src: typescript, alt: 'Typescript icon' },
];

const Home = (): ReactElement => {
  const classes = useStyles();

  function TechStack() {
    return (
      <div className={classes.techStack}>
        {techStackIcon.map((icon) => {
          return (
            <Box key={icon.src} className={classes.techStackLogo}>
              <img src={icon.src} alt={icon.alt} />
            </Box>
          );
        })}
        <Paper className={classes.reactLogo}>
          <img src={reactLogo} alt="ReactLogo" />
        </Paper>
        <Paper className={classes.reduxLogo}>
          <img src={reduxLogo} alt="ReduxLogo" />
        </Paper>
        <Box className={classes.techStackLogo}>
          <img src={aws} alt="AWS Logo" />
        </Box>
      </div>
    );
  }

  return (
    <Container
      component="main"
      disableGutters
      maxWidth={false}
      className={classes.container}
    >
      <Container
        component={Grid}
        container
        justifyContent="center"
        maxWidth="sm"
        className={classes.parent}
      >
        <Typography variant="h1">About Me</Typography>
        <Grid item xs={6}>
          <Typography variant="h5" className={classes.flip}>
            About Me
          </Typography>
        </Grid>
      </Container>
      <TechStack />
      <Container className={classes.mainContent}>
        <Typography>
          Welcome to my little home travaller. I really appreciate your coming.
        </Typography>
        <br />
        <Typography>
          I am a newbie who just graduated from Mohawk College.
        </Typography>
        <Typography>
          But I already got a degree in mathematics from the University of
          Waterloo, which is equivalent to computer science.
        </Typography>
        <br />
        <Typography>
          I am a person who is very interested in technology.
        </Typography>
        <Typography>Not just programing, but also hardware.</Typography>
        <Typography>
          Such as Loading OpenWrt firmware into my router and write scripts on
          it.
        </Typography>
        <br />
        <Typography>
          In software aspect, I am currently developing a freelance website,
          which is customized for the team leader. He is a graduate major in
          design.
        </Typography>
        <br />
        <Typography>
          I have a certain talent for algorithms, data structures, and design
          patterns.
        </Typography>
        <br />
        <Typography>
          The most advanced CS courses I took are distribute system, security,
          and the computer network.
        </Typography>
        <br />
        <br />
        <Typography>Learning</Typography>
        <br />
        <Typography>Enthusiasm</Typography>
        <br />
        <Typography>Focus</Typography>
      </Container>
      <div className={classes.ending}></div>
    </Container>
  );
};

export default Home;
