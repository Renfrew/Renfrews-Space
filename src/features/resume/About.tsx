import React, { ReactElement } from 'react';
import { Theme, makeStyles, createStyles } from '@material-ui/core/styles';

import { Button, Container, Grid, Typography } from '@material-ui/core';

const useStyles = makeStyles(({ breakpoints, spacing }: Theme) =>
  createStyles({
    container: {
      height: '100vh',
      display: 'flex',
      alignItems: 'center',
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
    },
    parent: {
      marginBottom: spacing(20),
      [breakpoints.down('sm')]: {
        paddingLeft: '10%',
      },
      [breakpoints.up('sm')]: {
        paddingLeft: breakpoints.values.sm * 0.1 + 'px',
      },
    },
    text: {
      color: 'white',
      mixBlendMode: 'difference',
      filter: 'drop-shadow(.05em .05em #cf0c0c)',
    },
    explore: {
      textAlign: 'center',
      marginTop: spacing(6),
      [breakpoints.down('sm')]: {
        marginRight: '10%',
      },
      [breakpoints.up('sm')]: {
        marginRight: breakpoints.values.sm * 0.1 + 'px',
      },
    },
  })
);

const Home = (): ReactElement => {
  const classes = useStyles();

  return (
    <Container
      component="main"
      disableGutters
      maxWidth={false}
      className={classes.container}
    >
      <Container
        component={Grid}
        container
        justifyContent="center"
        maxWidth="sm"
        className={classes.parent}
      >
        <Typography variant="h1" className={classes.text}>
          About Me
        </Typography>
        <Typography variant="h3" className={classes.text}>
          (About Me)
        </Typography>
        <Typography variant="h6" className={classes.text}>
          About Me
        </Typography>
        <Grid className={classes.explore}>
          <Button variant="contained" color="secondary" size="large">
            Explore
          </Button>
        </Grid>
      </Container>
    </Container>
  );
};

export default Home;
