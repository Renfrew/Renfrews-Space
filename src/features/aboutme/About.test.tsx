import React from 'react';
import renderer from 'react-test-renderer';
import About from './About';

describe('Testing About Page', () => {
  it('The NotFound page renders correctly', () => {
    const tree = renderer.create(<About />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
