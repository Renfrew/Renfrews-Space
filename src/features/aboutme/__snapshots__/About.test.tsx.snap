// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing About Page The NotFound page renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-disableGutters makeStyles-container-1 css-q7bdj4-MuiContainer-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiContainer-root MuiContainer-maxWidthSm makeStyles-parent-3 css-1vko90e-MuiGrid-root-MuiContainer-root"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-o2w69a-MuiTypography-root"
    >
      About Me
    </h1>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-8ngd37-MuiGrid-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-flip-2 css-ag7rrr-MuiTypography-root"
      >
        About Me
      </h5>
    </div>
  </div>
  <div
    className="makeStyles-techStack-4"
  >
    <div
      className="makeStyles-techStackLogo-5 css-1a8w37c"
    >
      <img
        alt="HTML5 icon"
        src="html5.png"
      />
    </div>
    <div
      className="makeStyles-techStackLogo-5 css-1a8w37c"
    >
      <img
        alt="CSS icon"
        src="css-512.png"
      />
    </div>
    <div
      className="makeStyles-techStackLogo-5 css-1a8w37c"
    >
      <img
        alt="Javascript icon"
        src="512px-JavaScript-logo.png"
      />
    </div>
    <div
      className="makeStyles-techStackLogo-5 css-1a8w37c"
    >
      <img
        alt="Typescript icon"
        src="typescript.png"
      />
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-reactLogo-6 css-1ps6pg7-MuiPaper-root"
    >
      <img
        alt="ReactLogo"
        src="react.png"
      />
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-reduxLogo-7 css-1ps6pg7-MuiPaper-root"
    >
      <img
        alt="ReduxLogo"
        src="Redux.png"
      />
    </div>
    <div
      className="makeStyles-techStackLogo-5 css-1a8w37c"
    >
      <img
        alt="AWS Logo"
        src="aws_icon_146074.png"
      />
    </div>
  </div>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg makeStyles-mainContent-8 css-cnms5y-MuiContainer-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      Welcome to my little home travaller. I really appreciate your coming.
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      I am a newbie who just graduated from Mohawk College.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      But I already got a degree in mathematics from the University of Waterloo, which is equivalent to computer science.
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      I am a person who is very interested in technology.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      Not just programing, but also hardware.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      Such as Loading OpenWrt firmware into my router and write scripts on it.
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      In software aspect, I am currently developing a freelance website, which is customized for the team leader. He is a graduate major in design.
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      I have a certain talent for algorithms, data structures, and design patterns.
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      The most advanced CS courses I took are distribute system, security, and the computer network.
    </p>
    <br />
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      Learning
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      Enthusiasm
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
    >
      Focus
    </p>
  </div>
  <div
    className="makeStyles-ending-9"
  />
</main>
`;
